### PR TITLE
Remove samplers for segmentation

### PR DESCRIPTION
**Pull request by JihwanEom**
## This PR includes:
- Delete `otx/mpa/modules/samplers` folder, it is not used anymore for all segmentation templates.
- Update legacies related with `MaxPoolingSampler`
- Remove `_mpl` build code lines in `setup.py`

### CODE REVIEW

**Review by goodsong81 — CHANGES_REQUESTED** [review]
Thank you for quick update!
Could you remove torch requirement from pyproject.toml?
One more question is, how's the accuracy or efficiency impacts?

File: pyproject.toml [hunk 1]
```diff
@@ -8,7 +8,7 @@ requires = [
     "wheel",
     "Cython~=0.29.32",
     "numpy~=1.21.0",
-    "torch@https://download.pytorch.org/whl/cu116/torch-1.13.1%2Bcu116-cp38-cp38-linux_x86_64.whl"
+    "torch==1.13.1"
```
Comment (goodsong81) [inline pyproject.toml]: It might be misleading. Could you remove this line?
This is build requirement, and no need if we remove mpl extension.

**Comment by JihwanEom** [comment]
> Thank you for quick update! Could you remove torch requirement from pyproject.toml? One more question is, how's the accuracy or efficiency impacts?

Here are experiment results :) I believe the values are in random seed range.

```
Dataset: kvasir-instrument
Lite-HRNet-18-mod2-baseline | mDice: 0.966 | time_per_epoch: 25.28
Lite-HRNet-s-mod2-baseline  | mDice: 0.958 | time_per_epoch: 17.83
Lite-HRNet-x-mod3-baseline  | mDice: 0.963 | time_per_epoch: 108.09

Lite-HRNet-18-mod2-maxpooling | mDice: 0.968 | time_per_epoch: 25.02
Lite-HRNet-s-mod2-maxpooling  | mDice: 0.961 | time_per_epoch: 18.25
Lite-HRNet-x-mod3-maxpooling  | mDice: 0.968 | time_per_epoch: 107.84
```

File: pyproject.toml (hunk 1 above)
Comment (JihwanEom) [reply pyproject.toml]: Thank you for quick feedback! Reflected.

**Review by goodsong81 — APPROVED** [review]

**Comment by goodsong81** [comment]
@JihwanEom I think I can merge this PR to mine. What do you think?

**Comment by JihwanEom** [comment]
> @JihwanEom I think I can merge this PR to mine. What do you think?

Absolutely agree with you :)

**Review by harimkang — COMMENTED** [review]
Is there no need to modify the `tox.ini` file in this PR? Doesn't this affect the test?

**Comment by goodsong81** [comment]
> Is there no need to modify the `tox.ini` file in this PR? Doesn't this affect the test?

Deps in tox are prerequisites. Deps in pyproject.toml is needed for building wheels from the project. So in tox run, it installs tox deps, and installs (in isolated env) pyproject.toml just for building wheel, then OTX is installed using the wheel.
In sum, this change will prevent tox from install torch twice. :)

But I'm not 100% sure;; I'll fix it if there is any problem.

**Comment by JihwanEom** [comment]
Segmentation local e2e test results:
4 failed cases found which are related with NNCF FQ.
```
TestToolsMPASegmentation.test_nncf_validate_fq[Custom_Semantic_Segmentation_Lite-HRNet-18-mod2_OCR] 
AssertionError: Incorrect number of FQs in optimized model: 580 != -1
TestToolsMPASegmentation.test_nncf_validate_fq[Custom_Semantic_Segmentation_Lite-HRNet-s-mod2_OCR] 
AssertionError: Incorrect number of FQs in optimized model: 432 != -1
TestToolsMPASegmentation.test_nncf_validate_fq[Custom_Semantic_Segmentation_Lite-HRNet-x-mod3_OCR] 
AssertionError: Incorrect number of FQs in optimized model: 1130 != -1
TestToolsMPASegmentation.test_nncf_validate_fq[Custom_Semantic_Segmentation_Lite-HRNet-18_OCR] 
AssertionError: Incorrect number of FQs in optimized model: 580 != -1
```
image

**Pull request merged**